src/command_modules/azure-cli-redis/azure/cli/command_modules/redis/_params.py

```diff
@@ -3,43 +3,54 @@
     from azure.cli.command_modules.redis._validators import JsonString, ScheduleEntryList
     from azure.cli.core.commands.parameters import get_enum_type  # TODO: Move this into Knack
-    from azure.cli.core.commands.parameters import get_resource_name_completion_list, name_type, tags_type
+    from azure.cli.core.commands.parameters import get_resource_name_completion_list
 
     with self.argument_context('redis') as c:
-        c.argument('name', options_list=['--name', '-n'], help='Name of the Redis cache.', id_part='name',
-                   completer=get_resource_name_completion_list('Microsoft.Cache/redis'))
+        cache_name = CLIArgumentType(options_list=['--name', '-n'], help='Name of the Redis cache.', id_part='name',
+                                     completer=get_resource_name_completion_list('Microsoft.Cache/redis'))
+        # pylint:disable=line-too-long
+        cache_name_without_id_part = CLIArgumentType(options_list=['--name', '-n'], help='Name of the Redis cache.', id_part=None,
+                                                     completer=get_resource_name_completion_list('Microsoft.Cache/redis'))
+        format_type = CLIArgumentType(options_list=['--file-format'], help='Format of the blob (Currently rdb is the only supported format, with other formats expected in the future)')
+
+        c.argument('name', arg_type=cache_name)
         c.argument('redis_configuration', help='JSON encoded configuration settings. Use @{file} to load from a file.',
                    type=JsonString)
         c.argument('reboot_type', arg_type=get_enum_type(RebootType))
         c.argument('key_type', arg_type=get_enum_type(RedisKeyType))
         c.argument('shard_id', type=int)
         c.argument('sku', help='Type of Redis cache.', arg_type=get_enum_type(SkuName))
         c.argument('vm_size', help='Size of Redis cache to deploy. Example : values for C family (C0, C1, C2, C3, C4, '
-                                   'C5, C6). For P family (P1, P2, P3, P4)')
+                                   'C5, C6). For P family (P1, P2, P3, P4, P5)')
         c.argument('enable_non_ssl_port', action='store_true')
         c.argument('shard_count', type=int)
-        c.argument('tags', tags_type)
-
-    for scope in ['redis import-method', 'redis import']:
-        with self.argument_context(scope) as c:
-            c.argument('files', nargs='+', help='Space-separated list of files to import.')
-            c.argument('file_format', help='File format to import.')
-
-    with self.argument_context('redis patch-schedule set') as c:
-        c.argument('schedule_entries', type=ScheduleEntryList)
-
-    with self.argument_context('redis create') as c:
-        c.argument('name', arg_type=name_type, completer=None)
+        c.argument('files', help='SAS url for blobs that needs to be imported', nargs='+')
+        c.argument('format', arg_type=format_type)
+        c.argument('file_format', arg_type=format_type)
```

Comment:
File_format and format are two different parameter destinations with the same argument. Are both destinations being used?

Reply:
yes